proto/profile.proto

```diff
@@ -115,6 +120,36 @@ message Sample {
   // label includes additional context for this sample. It can include
   // things like a thread id, allocation size, etc
   repeated Label label = 3;
+  // Breakdown can hold the individual timestamps and values for a sample that
+  // has aggregated values. The type and unit of each breakdown is defined by
+  // the corresponding entry in Profile.sample_type. It's acceptable to have
+  // less, but not more, breakdown entries than Profile.sample_type entries.
+  repeated Breakdown breakdown = 4;
+}
+
+// Breakdown stores data about the individual events that have been aggregated
+// into a Sample. There can be one breakdown for each Profile.sample_type.
+//
+// TODO: Figure out if this can also be used for the exemplar UC from alexey:
+// https://github.com/google/pprof/pull/725#issuecomment-1264841904
+message Breakdown {
+  // A list of timestamps at which the values of the associated sample were
+  // collected. The time is given in Profile.tick_unit resolution relative to
+  // Profile.time_nanos and values are delta encoded. So for three events
+  // captured at 100, 110 and 150 ticks after the profile start, this list should
+  // hold the values [100, 10, 40].
+  //
+  // TODO: Is delta encoding going to safe enough space to justify the
+  // complexity?
```

Comment:
This depends on the time range of the profile and the resolution of the unit. For instance, there are 60,000,000,000 nanoseconds in a minute. I'm guessing an average profile with nanosecond precision would benefit from delta encoding.

As for it being a hassle, I initially thought it would be. However, I think this approach is simple enough:

1. Since we have the breakdown now, we don't have to aggregate the `sample.value[type]` on every new sample. Instead, record the breakdown and use the absolute time when doing so, not the delta encoded one.
2. Before serializing, perform a sum on the `sample.breakdown[type].value[..]`s and put the aggregated value into `sample.value[type]`. In the same outer loop you can also delta-encode the `sample.breakdown[type].ticks[..]`.

Seems viable?

Reply:
Thanks for the thoughtful comment on this. I've already done some testing (based on an earlier profile.proto change idea) and found that delta encoding didn't have a significant impact. I'm also going to write down some details why I think this is the case later today.  As for as complexity is concerns: I agree, it's not too bad, but I had some backchannel discussions with the pprof folks and they want to keep things simple if possible. FWIW I emphasize with that. Once you start putting custom encoding schemes on top of protocol buffers, you lose some of their value. At some point it might even make sense to ask if the overhead of protocol buffers is acceptable (each value gets its own little "header" tag).

Anyway, I haven't made up my mind on delta yet. But will share some more data points later on to help the discussion.

Reply:
I made a TODO in the PR description to share some benchmarks on it. Resolving this thread for now.